pkg/registry/generic/registry/storage_factory.go

```diff
@@ -30,7 +30,8 @@ func StorageWithCacher(
 	objectType runtime.Object,
 	resourcePrefix string,
 	scopeStrategy rest.NamespaceScopedStrategy,
-	newListFunc func() runtime.Object) storage.Interface {
+	newListFunc func() runtime.Object,
+	indices ...storage.IndexingFunc) storage.Interface {
```

Comment:
With the new interface, I agree with that.

However, I would like to leave as is for now (to avoid changing all files to pass this in this PR (which is definitely non trivial)) and change that in the follow-up PR (added a TODO). Does it sound ok?